### PR TITLE
feat: add environment to navbar switcher

### PR DESCRIPTION
**Pull request by massao**
# Purpose of PR

Add environment name to the navbar switcher

### Preview
image

### CODE REVIEW

File: packages/components/navbar/src/NavbarSwitcher/NavbarSwitcher.tsx [hunk 1]
```diff
@@ -41,35 +45,90 @@ function splitSpaceName(
...
+      <span>{start}</span>
+      {middle && (
+        <span className={styles.switcherSpaceNameTruncation}>{middle}</span>
+      )}
+      {end && <span>{end}</span>}
+    </Text>
+  ) : (
+    <Text fontColor={isEnv ? envColor : 'gray600'}>{value}</Text>
+  );
+};
```
Comment (cf-remylenoir) [inline packages/components/navbar/src/NavbarSwitcher/NavbarSwitcher.tsx:114]: Could we consolidate colors like so?

```suggestion
const SwitcherLabel = ({
  value,
  styles,
  envVariant,
  type,
}: SwitcherLabelProps) => {
  const [start, middle, end] =
    typeof value === 'string' ? splitString(value) : [];
  const isEnv = type === 'environment';
  const isMaster = envVariant === 'master';
  const fontColor = isEnv && isMaster ? 'green600' : 'gray600';
  
  return start !== undefined ? (
    <Text fontColor={fontColor}>
      <span>{start}</span>
      {middle && (
        <span className={styles.switcherSpaceNameTruncation}>{middle}</span>
      )}
      {end && <span>{end}</span>}
    </Text>
  ) : (
    <Text fontColor={fontColor}>{value}</Text>
  );
};
```

Comment (cf-remylenoir) [reply packages/components/navbar/src/NavbarSwitcher/NavbarSwitcher.tsx:114]: Or rather

```typescript
switch(true) {
  case isEnv && isMaster:
    return 'green600';
  case isEnv && !isMaster || isEnv:
  default:
    return 'gray600';
}
```

File: packages/components/navbar/src/NavbarSwitcher/NavbarSwitcher.tsx [hunk 2]
```diff
@@ -41,35 +45,98 @@ function splitSpaceName(
...
-  children: ReactNode;
-  isCircle?: boolean;
-  envVariant?: 'master' | 'non-master';
-  isAlias?: boolean;
-  ariaLabel?: string;
-};
+type NavbarLoadingProps =
+  | {
+      isLoading?: true;
+      environemnt?: never;
```
Comment (denkristoffer) [inline packages/components/navbar/src/NavbarSwitcher/NavbarSwitcher.tsx:51]:
```suggestion
      environment?: never;
```

File: packages/components/navbar/src/NavbarSwitcher/NavbarSwitcher.tsx [hunk 3]
```diff
@@ -41,35 +45,98 @@ function splitSpaceName(
...
+      space?: never;
+    }
+  | {
+      isLoading?: false;
+      environemnt?: string;
```
Comment (denkristoffer) [inline packages/components/navbar/src/NavbarSwitcher/NavbarSwitcher.tsx:56]:
```suggestion
      environment?: string;
```

File: packages/components/navbar/src/NavbarSwitcher/NavbarSwitcher.tsx [hunk 4]
```diff
@@ -41,35 +45,98 @@ function splitSpaceName(
...
+    typeof value === 'string' ? splitString(value) : [];
+  const envColor = useMemo(() => {
+    const isEnv = type === 'environment';
+    const isMaster = envVariant === 'master';
+    switch (true) {
+      case isEnv && isMaster:
+        return 'green600';
+      default:
+        return 'gray600';
+    }
```
Comment (denkristoffer) [inline packages/components/navbar/src/NavbarSwitcher/NavbarSwitcher.tsx:108]: Can we do if statements and not try to be overly clever?

File: packages/components/navbar/src/NavbarSwitcher/NavbarSwitcher.tsx [hunk 5]
```diff
@@ -41,35 +45,98 @@ function splitSpaceName(
...
+      space?: string;
+    };
+
+type NavbarSwitcherOwnProps = CommonProps &
+  NavbarLoadingProps & {
+    isCircle?: boolean;
+    envVariant?: 'master' | 'non-master';
+    isAlias?: boolean;
+    ariaLabel?: string;
+    environment?: React.ReactNode;
```
Comment (denkristoffer) [inline packages/components/navbar/src/NavbarSwitcher/NavbarSwitcher.tsx:66]: Does this type still make sense with the new `NavbarLoadingProps`?

File: packages/components/navbar/src/NavbarSwitcher/NavbarSwitcher.tsx (hunk 4 above)
Comment (cf-remylenoir) [reply packages/components/navbar/src/NavbarSwitcher/NavbarSwitcher.tsx]: Sorry @massao 😄

**Review by cf-remylenoir — APPROVED** [review]
It looks good to me, my only remark would be related to the poor contrast of the caret icon color with its background. @damann what do you think, does it satisfy an accessible contrast ratio?

**Comment by damann** [comment]
> It looks good to me, my only remark would be related to the poor contrast of the caret icon color with its background. @damann what do you think, does it satisfy an accessible contrast ratio?

Let's bump it up to grey500 (contrast ration 4.57:1)

**Review by damann — APPROVED** [review]

**Pull request merged**